FHIR/sushi
# Check exported resources for presence of multiple choice values

**Description:**
When a choice element is defined in FHIR, it allows for values of different types with names based on those types. However, any choice element that is present on resource should have only one named value. For example, Observation.value[x] allows for many types, but an Observation resource should not have both valueString and valueInteger. After exporting a resource, check the resource for the presence of multiple values for the same choice element, and log an error when they are detected.

As assigned values are validated in exporters, the relevant StructureDefinitions will have their elements unfolded. Keep the same instance of that StructureDefinition that was used for validation so that the unfolded elements are present when performing the multiple choice validation check.

**Testing Instructions:**
Confirm that all tests pass. Confirm that tests provide sufficient coverage for the scenario described. I put most of the tests in `InstanceExporter.test.ts` since those were the first ones I wrote, but since all exporters use the same functions, I felt that it was not necessary to have as many tests in the other exporters.

**Related Issue:**
Fixes #1137.
